### PR TITLE
Closes #4044 - Adds ability to register non-standard views as part of a NautobotUIViewSet

### LINKED ISSUE #4044: Ability to register additional method to a UI Viewset via NautobotUIViewSetRouter
### As ...

P.D. - Plugin Developer

### I want ...

The ability to define additional routes as part of a single viewset for a given model

### So that ...

I can build one call to handle all out of the box views plus any additional non-standard views.

### I know this is done when...

- I create a new model in a plugin that requires N non-standard views and each non-standard view does NOT require an additional URL pattern to be registered.
- 

### Optional - Feature groups this request pertains to.

- [ ] Automation
- [ ] Circuits
- [ ] DCIM
- [ ] IPAM
- [ ] Misc (including Data Sources)
- [ ] Organization
- [X] Plugins (and other Extensibility)
- [ ] Security (Secrets, etc)
- [ ] Image Management
- [ ] UI/UX
- [ ] Documentation
- [ ] Other (not directly a platform feature)

### Database Changes

None

### External Dependencies

None

### PR DESCRIPTION
**Pull request by whitej6**
# Closes: #4044 
# What's Changed

- Adds ability to add additional `GET` detail views to a NautobotUIViewSet via the `@action(detail=True)` decorator
- Moved `notes` and `changelog` views to use the DynamicRoute

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/core/views/mixins.py [hunk 1]
```diff
@@ -957,6 +958,7 @@ class ObjectChangeLogViewMixin(NautobotViewSetMixin):
 
     base_template = None
 
+    @rf_action(detail=True)
```
Comment (gsnider2195) [inline nautobot/core/views/mixins.py:961]: Might be more clear to import this as `drf_action`

Comment (whitej6) [reply nautobot/core/views/mixins.py:961]: Good point, will swap out.

**Review by timizuoebideri1 — APPROVED** [review]

**Review by bryanculver — CHANGES_REQUESTED** [review]
Docs, please! Otherwise looks good.

**Comment by whitej6** [comment]
@bryanculver docs added, referenced v1.6.0 as version introduced.

**Review by glennmatthews — APPROVED** [review]

**Review by bryanculver — APPROVED** [review]

**Pull request merged**